homeassistant/components/tessie/coordinator.py

```diff
@@ -68,18 +84,61 @@ async def _async_update_data(self) -> dict[str, Any]:
                 raise ConfigEntryAuthFailed from e
             raise
 
-        return self._flatten(vehicle)
-
-    def _flatten(
-        self, data: dict[str, Any], parent: str | None = None
-    ) -> dict[str, Any]:
-        """Flatten the data structure."""
-        result = {}
-        for key, value in data.items():
-            if parent:
-                key = f"{parent}_{key}"
-            if isinstance(value, dict):
-                result.update(self._flatten(value, key))
-            else:
-                result[key] = value
-        return result
+        return flatten(vehicle)
+
+
+class TessieEnergySiteLiveCoordinator(DataUpdateCoordinator[dict[str, Any]]):
```

Comment:
Have you considered creating a base coordinator? I have this in `withings`

Reply:
I did this previously in Teslemetry but it caused a bunch of typing issues and there really wasn't that much overlap. It ended up adding complexity rather than removing it.